src/filter/index.js

```diff
@@ -17,6 +19,13 @@ export default async function filter(content, selector, filterNames, filterFunct
     });
   }
 
+  Array.from(webPageDOM.querySelectorAll('a')).map(link => {
```

Comment:
Improve selector instead of checking if the href is defined, like this `'a:not([href=""])'`.